onnxruntime/core/providers/coreml/coreml_execution_provider.cc

```diff
@@ -36,6 +37,154 @@ CoreMLExecutionProvider::CoreMLExecutionProvider(const CoreMLOptions& options)
 
 CoreMLExecutionProvider::~CoreMLExecutionProvider() {}
 
+// Create a helper function for type checking
+bool CoreMLExecutionProvider::IsSupportedType(int32_t type) const {
+  return supported_input_output_types_.find(type) != supported_input_output_types_.end();
+}
+
+// After an input node is removed from the partition, we need to update the inputs of the partition.
+// We evaluate the outputs of the input nodes
+// For each output, if its consumer node is part of the partition, we add this output to the inputs of the partition
+void UpdateInputsSetAfterNodeRemoval(std::unordered_set<NodeIndex>& partition_nodes_set,
+                                     const Node* node,
+                                     std::unordered_set<std::string>& newInputs,
```

Comment:
nit: naming convention - `new_inputs` vs `newInputs`. same thing for `newOutputs` below.